nucleus/common/common-util/src/main/java/com/sun/enterprise/util/io/FileUtils.java

```diff
@@ -373,21 +370,7 @@ public static String makeFriendlyFilenameExtension(String filename) {
 
         filename = makeLegalNoBlankFileName(filename);
 
-        String extension = "";
-        if (filename.endsWith(".ear")) {
-            filename = filename.substring(0, filename.indexOf(".ear"));
-            extension = "_ear";
-        } else if (filename.endsWith(".war")) {
-            filename = filename.substring(0, filename.indexOf(".war"));
-            extension = "_war";
-        } else if (filename.endsWith(".jar")) {
-            filename = filename.substring(0, filename.indexOf(".jar"));
-            extension = "_jar";
-        } else if (filename.endsWith(".rar")) {
-            filename = filename.substring(0, filename.indexOf(".rar"));
-            extension = "_rar";
-        }
-        return filename + extension;
+        return addExtentiontoFileName(filename);
```

Comment:
Compare the block replaced by `addExtentiontoFileName` with the method body. In the method that match is done with `filename.endsWith("_ear")` whereas the deleted section does `filename.endsWith(".ear")` and so forth for the other endings. This confuses me as it looks like a change of behaviour. Maybe you can explain why it is not?

Reply:
I think looking at file history when I merged master in to this branch something wasn't merged correctly, I have reverted all my changes to FileUtils and redone them.